README.md

```diff
@@ -83,10 +83,33 @@ During development in Picard, it is sometimes necessary to build locally against
 
 #### Releasing Picard
 
-Full instructions on how to create a new release of Picard are [here](https://github.com/broadinstitute/picard/wiki/How-to-release-Picard)
+Full instructions on how to create a new release of 
+Picard are [here](https://github.com/broadinstitute/picard/wiki/How-to-release-Picard)
 
-----
+#### Path providers
+
+Picard has limited support for reading from Path providers. Currently only google's api is supported, and only a few tools support this
+To run with this support you will need to download the google nio jar (e.g. from [maven](http://central.maven.org/maven2/com/google/cloud/google-cloud-nio/0.29.0-alpha/google-cloud-nio-0.29.0-alpha.jar))
+and then run picard as follows:
+
+```
+java -cp <PATH TO google-cloud-nio> -cp <PATH TO picard.jar> picard.cmdline.PicardCommandLine <Picard arguments starting from program>
+```
+For example:
+
+```
+java -cp google-cloud-nio.jar -cp picard.jar picard.cmdline.PicardCommandLine CrosscheckFingerprints \
+I=gs://sample1.vcf \
+I=gs://sample2.vcf \
+CROSSCHECK_BY=FILE \
+H=Haplotype_db.txt \
+O=crosscheck.out
+```
+
+Alternatively, you can run the tool via the [GATK](https://software.broadinstitute.org/gatk/download/) which bundles the Google-Cloud
+jar, and should thus "Just Work".
```

Comment:
Any way you can set some setting in gradle to bundle this in for you, but have it default to exclude this dependency (optional dependency)?

Reply:
I don't know gradle well enough.. @cmnbroad can you advise here?

Reply:
I think the gradle way would be to create a separate ```cloud``` configuration and task for the nio-aware artifact:

```
configurations {
    cloudConfiguration {
        extendsFrom runtime
    }
}

task cloudJar(type: com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar)  {
    configurations = [project.configurations.cloudConfiguration]
    from project.sourceSets.main.output
    archiveName 'picardcloud.jar'
}
```
Then add the conditional dependency to the list of dependencies (the configuration definition above must appear before this in the file):

```
    configurations.cloudConfiguration {
        compile ('com.google.cloud:google-cloud-nio:0.5.0')
    }
```

Then ```.gradlew cloudJar``` will create a picardcloud.jar that includes the, without affecting the other jars.

Reply:
thanks @cmnbroad. 
is there a reason you quoted `com.google.cloud:google-cloud-nio:0.5.0' as opposed to `com.google.cloud:google-cloud-nio:0.28.0-alpha:shaded` (which I think is the version we use)?

Reply:
No reason, I just took those coordinates off of maven as an example.